workflow/executor.go

```diff
@@ -313,23 +313,29 @@ func (e *executor) processSteps(
 	return runnableSteps, stepOutputProperties, stepLifecycles, stepRunData, nil
 }
 
-func applyLifecycleScopes(
-	stepLifecycles map[string]step.Lifecycle[step.LifecycleStageWithSchema],
-	typedInput schema.Scope,
-) error {
+// BuildNamespaces creates a namespaced collection of objects for the inputs
+// and outputs of each stage in the step's lifecycles.
```

Comment:
1. This doesn't really tell me much. Reading this I'd expect to see some description of the two levels of string map, because the type tells me nothing about what those strings mean.
2. Jared's original comment suggested "I think it would make more sense to talk about how it maps the step's path and ID to a map of object IDs to object schemas", and I don't really see any of that in here. 😦 Maybe it's just me...
3. If this is a capital letter name, shouldn't it be in the interface?